File: web_shop_backend/src/main/java/com/lap/web_shop_backend/model/Address.java
package com.lap.web_shop_backend.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Address {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    private String first_name;

    private String last_name;

    private String street;

    private String city;

    private String country;

    private Integer postal_code;

    @OneToMany(mappedBy = "billing_address")
    @JsonIgnore
    private List<Customer> customers;

    @OneToMany(mappedBy = "shipping_address")
    @JsonIgnore
    private List<Shipment> shipments;
}

File: web_shop_frontend/src/models/Address.ts
export default class Address {
    id: Number
    first_name: String
    last_name: String
    street: String
    city: String
    country: String
    postal_code: Number

    constructor(data: any) {
        this.id = data.id
        this.first_name = data.first_name
        this.last_name = data.last_name
        this.street = data.street
        this.city = data.city
        this.country = data.country
        this.postal_code = data.postal_code
    }
}